src/easyscore.js

```diff
@@ -271,12 +292,18 @@ export class EasyScore {
     this.options = Object.assign({
       factory: null,
       builder: null,
+      commitHooks: [
+        setId,
+        setClass,
+        Articulation.easyScoreHook,
```

Comment:
It's not problematic, but I think each element should have all it's logic, constants, hooks, etc. all in one place (yes, there's still lots of room for improvement in the existing code). I don't think the keyval options really need to be tightly coupled to EasyScore -- they're by design meant to allow for extensions. 

Also, I don't think that following the evolution of EasyScore is that useful (at least compared to having cohesive units that can be managed separately.)